usb.go

```diff
@@ -165,7 +165,27 @@ func newContextWithImpl(impl libusbIntf) *Context {
 
 // NewContext returns a new Context instance.
 func NewContext() *Context {
-	return newContextWithImpl(libusbImpl{})
+	return ContextOptions{}.New()
+}
+
+// DeviceDiscovery enables a scan of the available USB devices connected to the system
+type DeviceDiscovery int
+
+const (
+	EnableDeviceDiscovery = iota
+	DisableDeviceDiscovery
+)
+
+// ContextOptions holds optional flags checked when creating a new Context
+type ContextOptions struct {
+	DeviceDiscovery DeviceDiscovery
+}
+
+// New creates a Context, taking into account the optional flags contained in ContextOptions
+func (o ContextOptions) New() *Context {
+	return newContextWithImpl(libusbImpl{
+		discovery: o.DeviceDiscovery,
```

Comment:
for your consideration: we could also have discovery be a boolean, and set it to `o.DeviceDiscovery == EnableDeviceDiscovery` here. I'm not sure if it would be much better though, because on one hand it removes a bit of complexity and tight coupling from the libusbImpl, but on the other hand it moves it here and uses two mechanisms instead of one.

Reply:
Yes, correct, I was indeed thinking on what would be the cleanest way of passing that parameter 'deeper', but decided that converting the enum to a boolean would just make it more confusing.

What about having the `DeviceDiscovery` be a boolean instead of a "boolean enum" ?

Reply:
DeviceDiscovery as a boolean would default to discovery disabled with zero value. That means all users that want to use `ContextOptions` would need to always include a boilerplate DeviceDiscovery: true to get what should be the default behavior. An enum allows us to interpret zero value as enabled.
Another option is to use DisableDeviceDiscovery as you suggested initially, but that leads to double negatives.